### PR TITLE
Stats: Apply dropdown navigation for small screen on summary pages

### PR DESCRIPTION
**Pull request by grzegorz-cp**
#### Proposed Changes

* Creare new component for swapping different components at a specified breakpoint
* Swap tabs for a dropdown for small screens on a summary page

#### Testing Instructions

* Go to stats and navigate to `View all`/`View details` in a footer of a module below the chart (you will go to a summary detail page)
* apply the feature flag by adding `?flags=stats/horizontal-bars-everywhere` to the URL
* Resize the page and make sure that for a small screen, tabs are converted to a dropdown
* Bonus test: Make sure that the navigation doesn't break for different screen sizes

Widescreen header:
Screenshot 2023-01-25 at 7 29 34 PM

Mobile screen header:
Screenshot 2023-01-25 at 7 29 56 PM

Before the proposed changes, the widescreen header was applied to all breakpoints. This PR adds the dropdown visible on the mobile screenshot. 

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #71018

### CODE REVIEW

File: client/my-sites/stats/stats-module/all-time-nav.jsx [hunk 1]
```diff
@@ -71,23 +105,12 @@ export const StatsModuleSummaryLinks = ( props ) => {
...
-										selected={ i.value === selected.value }
-									>
-										{ i.label }
-									</SegmentedControl.Item>
-								);
-							} ) }
-						</SegmentedControl>
+						<ComponentSwapper
+							className={ classnames( 'stats-summary-nav__intervals-container' ) }
+							breakpint="<660px"
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-module/all-time-nav.jsx]:
```suggestion
							breakpoint="<660px"
```

File: packages/components/src/component-swapper/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+`ComponentSwapper` is a react component that can be used to swap components at a specified breakpoint. For example a buttons navigation to a dropdown for samll screens.
+
+## Properties
+
+### `className { string }`
+
+CSS class to be applied to a root div.
+
+### `breakpint { string } - default: '<660px'`
```
Comment (jsnmoon) [inline packages/components/src/component-swapper/README.md]:
```suggestion
### `breakpoint { string } - default: '<660px'`
```

File: packages/components/src/component-swapper/index.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { useBreakpoint } from '@automattic/viewport-react';
+import React, { ReactChild, useEffect } from 'react';
+
+const ComponentSwapper = ( {
+	className,
+	breakpint = '<660px', // breakpoints from mediaQueryLists from '@automattic/viewport'
```
Comment (jsnmoon) [inline packages/components/src/component-swapper/index.tsx]:
```suggestion
	breakpoint = '<660px', // breakpoints from mediaQueryLists from '@automattic/viewport'
```

File: packages/components/src/component-swapper/index.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	breakpointActiveComponent,
+	breakpointInactiveComponent,
+	onSwap,
+	children,
+}: {
+	children: ReactChild[];
+	breakpint: string;
```
Comment (jsnmoon) [inline packages/components/src/component-swapper/index.tsx]:
```suggestion
	breakpoint: string;
```

File: packages/components/src/component-swapper/index.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+	breakpointActiveComponent: React.Component;
+	breakpointInactiveComponent: React.Component;
+	onSwap?: () => void;
+	className?: string;
+} ) => {
+	const isActive = useBreakpoint( breakpint );
```
Comment (jsnmoon) [inline packages/components/src/component-swapper/index.tsx]:
```suggestion
	const isActive = useBreakpoint( breakpoint );
```

**Review by jsnmoon — APPROVED** [review]
This works exactly as described. I left code change suggestions to fix a typo, but nothing blocking.

What do you think about using ComponentSwapper to refactor code paths where we use DotPager? For example, I remember a nontrivial amount of duplicated code in the "All-time highlights" section on the insights page.

**Comment by grzegorz-cp** [comment]
> What do you think about using ComponentSwapper to refactor code paths where we use DotPager

That would be another great example of a similar pattern. I've created a maintenance issue to apply the component [#72605](https://github.com/Automattic/wp-calypso/issues/72605).

Thanks for noticing the typos.

**Review by kangzj — APPROVED** [review]
It works as described. Really nice refactoring! 👍 

I'm just thinking out aloud - nothing needs to be changed at the stage.
- If we have more than one breakpoint, what the swapper would look like.
- How we could deal with affection of the width of the sidebar.

**Comment by grzegorz-cp** [comment]
Great points. To incorporate the sidebar, we would have to extend the component to use a container query instead of a regular media query. To mitigate that for now, we could manually calculate a new breakpoint (add the sidebar width). But at the moment, the component only supports the set of breakpoints defined in `@automattic/viewport-react`. So we would have to make it more flexible first.

For multiple breakpoints, I guess we could theoretically nest the components since they accept `children`, but that's a great improvement idea. The `breakpoint` prop could be extended to accept a string or an array of values, and the component could handle it internally.

**Pull request merged**